src/main/scala/de/zalando/beard/parser/BeardTemplateListener.scala

```diff
@@ -107,12 +106,25 @@ class BeardTemplateListener extends BeardParserBaseListener {
 
   override def exitBeard(ctx: BeardContext) = {
     val statements = ctx.statement().map(_.result).toList
+
     val renderStatements = statements.collect {
       case renderStatement: RenderStatement => renderStatement
     }
-    ctx.result = statements
-    val extended = Option(ctx.extendsStatement()).map(_.result)
-    val contentForStatements = ctx.contentForStatement().map(_.result).toList
-    result = BeardTemplate(statements, extended, renderStatements, contentForStatements)
+
+    val extended = statements.collectFirst {
+      case extendsStatement: ExtendsStatement => extendsStatement
+    }
+
+    val contentForStatements = statements.collect {
+      case contentForStatement: ContentForStatement => contentForStatement
+    }
+
+    val finalStatements = statements.reverse.takeWhile {
```

Comment:
@danpersa do you think there is a better way of implementing this? What this code does is not to render content in between extends and contentFor statement.